code/datums/materials/_material.dm

```diff
@@ -129,6 +129,9 @@ Simple datum which is instanced once per type and is used for every object of sa
 	if(material_flags & MATERIAL_ADD_PREFIX)
 		source.name = initial(source.name)
 
+	if(beauty_modifier) //component/beauty/InheritComponent() will handle the removal.
+		INVOKE_ASYNC(source, /datum.proc/_AddComponent, list(/datum/component/beauty, -(beauty_modifier * amount)))
```

Comment:
why are you doing this async?

Reply:
A bandaid ninjanomnom suggested to floyd when he first added the component, to fix some unspecified issues they both forgot about now.